trace_decoder/src/decoding.rs

```diff
@@ -303,18 +307,20 @@ impl ProcessedBlockTrace {
     fn pad_gen_inputs_with_dummy_inputs_if_needed(
         gen_inputs: &mut Vec<TxnProofGenIR>,
         other_data: &OtherBlockData,
-        extra_data: &ExtraBlockData,
+        final_extra_data: &ExtraBlockData,
+        initial_extra_data: &ExtraBlockData,
         initial_tries: &PartialTrieState,
         final_tries: &PartialTrieState,
         has_withdrawals: bool,
     ) -> bool {
         match gen_inputs.len() {
             0 => {
                 debug_assert!(initial_tries.state == final_tries.state);
+                debug_assert!(initial_extra_data == final_extra_data);
```

Comment:
nit: Do you think it's worth adding a error string for these? Might be overkill idk.

Reply:
I think the assertions are actually not needed here, as it's solely based on gen_inputs len, meaning the loop wouldn't have been entered before